### PR TITLE
verification docs updating for the new snowtrace

### PR DESCRIPTION
**Pull request by Migl992**
# Docs PR Template

## Why this should be merged

-Updated the verification steps using hardhat
-Added verification trough APIs

## How this works

<please give a thorough description of what your changes are and how they work>

## How these changes were tested 

<please give a thorough description of how to view the changes and how they were tested>

## To prevent any errors while building

- [] run `vale /docs/file/path` on all changed `.md` files to ensure all grammar rules pass
- [] run `markdownlint /docs/file/path` on all changed `.md` files to ensure all linting rules pass
- [] complete the above two checks and all additional rules outlined in `style-checker-notes.md` to
  ensure all checks pass

### If a document was removed/deleted

- [] the path to that doc must be redirected to a valid URL via the
  `_redirects` file
- [] `_redirects` were manually verified with the cloudflare preview link
- [] `sidebars.json` reflects all changes made to file path

### If a document was moved

- [] all files that were moved from their current directory to a new path have had their paths
  redirected via the `_redirects` file
- [] `_redirects` were manually verified with the cloudflare preview link
- [] `sidebars.json` reflects all changes made to file path

### CODE REVIEW

File: docs/build/dapp/smart-contracts/verification/verify-hardhat.md [hunk 1]
```diff
@@ -64,23 +67,51 @@ menu labeled “Optimization”
    1. If successful, all Contracts with the same bytecode will be verified
    2. If unsuccessful, read the error messages provided and make the appropriate changes
       1. Ensure to check that the compiler version and optimizer runs are the same as when you 
-      compiled  the contract prior to deployment
+      compiled  the contract prior to deployment -->
+## Verify the Smart Contract programmatically using APIs
```
Comment (meaghanfitzgerald) [inline docs/build/dapp/smart-contracts/verification/verify-hardhat.md:71]:
```suggestion
## Verify the Smart Contract Programmatically Using APIs
```

File: docs/build/dapp/smart-contracts/verification/verify-hardhat.md [hunk 2]
```diff
@@ -64,23 +67,51 @@ menu labeled “Optimization”
...
+
+Ensure you have Postman or any other API platform installed on your computer (or accessible through online services), 
+along with your contract's source code and the parameters utilized during deployment.
+
+Here is the API call URL to use for a POST request: 
+
+```https://api.snowtrace.io/api?module=contract&action=verifysourcecode ```
+
+Please note that this URL is specifically configured for mainnet verification on the Avalanche c-chain.
```
Comment (meaghanfitzgerald) [inline docs/build/dapp/smart-contracts/verification/verify-hardhat.md:80]:
```suggestion
Please note that this URL is specifically configured for Mainnet verification on the Avalanche c-chain. 
```

File: docs/build/dapp/smart-contracts/verification/verify-hardhat.md [hunk 3]
```diff
@@ -30,8 +30,11 @@ Some clean-up may be necessary to get the code to compile properly in the Snowtr
   - If the contract uses multiple SPDX licenses, use both licenses by adding AND:
     `SPDX-License-Identifier: MIT AND BSD-3-Clause`
 
-## Verify the Smart Contract using Snowtrace
+## Verify the Smart Contract using Snowtrace UI
 
+Snowtrace is currently working on a new user interface (UI) for smart contract verification. 
+Meanwhile, consider utilizing our API for a seamless smart contract verification experience.
```
Comment (ashucoder9) [inline docs/build/dapp/smart-contracts/verification/verify-hardhat.md:36]:
```suggestion
Meanwhile, you may consider using their API for a seamless smart contract verification experience.
```

File: docs/build/dapp/smart-contracts/verification/verify-hardhat.md [hunk 4]
```diff
@@ -64,23 +67,51 @@ menu labeled “Optimization”
...
-      compiled  the contract prior to deployment
+      compiled  the contract prior to deployment -->
+## Verify the Smart Contract Programmatically Using APIs
+
+Ensure you have Postman or any other API platform installed on your computer (or accessible through online services), 
+along with your contract's source code and the parameters utilized during deployment.
+
+Here is the API call URL to use for a POST request: 
+
+```https://api.snowtrace.io/api?module=contract&action=verifysourcecode ```
```
Comment (ashucoder9) [inline docs/build/dapp/smart-contracts/verification/verify-hardhat.md:78]:
```suggestion
```https://api.snowtrace.io/api?module=contract&action=verifysourcecode```
```

File: docs/build/dapp/smart-contracts/verification/verify-hardhat.md [hunk 5]
```diff
@@ -64,23 +67,51 @@ menu labeled “Optimization”
...
+
+Please note that this URL is specifically configured for Mainnet verification on the Avalanche C-Chain.
```
Comment (ashucoder9) [inline docs/build/dapp/smart-contracts/verification/verify-hardhat.md:80]:
```suggestion
Please note that this URL is specifically configured for verifying contracts on the Avalanche C-Chain Mainnet. 
```

File: docs/build/dapp/smart-contracts/verification/verify-hardhat.md [hunk 6]
```diff
@@ -64,23 +67,51 @@ menu labeled “Optimization”
...
+
+Ensure you have Postman or any other API platform installed on your computer (or accessible through online services), 
+along with your contract's source code and the parameters utilized during deployment.
+
+Here is the API call URL to use for a POST request: 
+
+```https://api.snowtrace.io/api?module=contract&action=verifysourcecode ```
+
+Please note that this URL is specifically configured for Mainnet verification on the Avalanche C-Chain. 
+If you intend to verify on the testnet use:
```
Comment (ashucoder9) [inline docs/build/dapp/smart-contracts/verification/verify-hardhat.md:81]:
```suggestion
If you intend to verify on the Fuji Testnet, use:
```

File: docs/build/dapp/smart-contracts/verification/verify-hardhat.md [hunk 7]
```diff
@@ -64,23 +67,51 @@ menu labeled “Optimization”
...
 
 ## Verifying with Hardhat-Verify
 
 This part of the tutorial assumes that the contract was deployed using Hardhat and that all Hardhat 
 dependencies are properly installed to include `'@nomiclabs/hardhat-etherscan'`.
 
-You will need to create a `.env.json` with your _Wallet Seed Phrase_ and _Snowtrace API key_
-
-You will need to obtain an _API key_ [here](https://snowtrace.io/myapikey)
+You will need to create a `.env.json` with your _Wallet Seed Phrase_, you don't need an API key to verify on Snowtrace
```
Comment (ashucoder9) [inline docs/build/dapp/smart-contracts/verification/verify-hardhat.md:108]:
```suggestion
You will need to create a `.env.json` with your _Wallet Seed Phrase_. You don't need an API key to verify on Snowtrace.
```

**Comment by meaghanfitzgerald** [comment]
@Migl992 appreciate the contribution

**Pull request merged**